types/terser-webpack-plugin/index.d.ts

```diff
@@ -0,0 +1,42 @@
+// Type definitions for terser-webpack-plugin 1.2
+// Project: https://github.com/webpack-contrib/terser-webpack-plugin
+// Definitions by: Daniel Schopf <https://github.com/Danscho>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+import { Plugin } from 'webpack';
+import { MinifyOptions } from 'terser';
+import webpack = require('webpack');
+
+export = TerserPlugin;
+
+declare class TerserPlugin extends Plugin {
+    constructor(opts?: TerserPlugin.TerserPluginOptions);
+}
+
+declare namespace TerserPlugin {
+    interface MinifyResult {
+        error: any;
+        map: any;
+        code: any;
+        warnings: any;
+        extractedComments: any;
+    }
+
+    type ExtractCommentFn = (node: any, comment: any) => (boolean | object);
+
+    interface TerserPluginOptions {
+        test?: string | RegExp | Array<string | RegExp>;
+        include?: string | RegExp | Array<string | RegExp>;
+        exclude?: string | RegExp | Array<string | RegExp>;
+        chunkFilter?: (chunk: webpack.compilation.Chunk) => boolean;
+        cache?: boolean | string;
+        cacheKeys?: (defaultCacheKeys: any, file: any) => object;
+        parallel?: boolean | number;
+        sourceMap?: boolean;
+        minify?: (file: any, sourceMap: any) => MinifyResult;
+        terserOptions?: MinifyOptions;
+        extractComments?: boolean | string | RegExp | object | ExtractCommentFn;
```

Comment:
this can be more specific than object with `{ condition: Boolean | string| RegExp | ExtractCommentFn|object; filename?: string | (file: string) => string; banner?: Boolean|string|(banner: string) => string;}`